homeassistant/components/bond/config_flow.py

```diff
@@ -42,32 +47,73 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
 
-    async def async_step_user(self, user_input=None):
-        """Handle the initial step."""
+    _discovered: dict = None
+
+    async def async_step_zeroconf(
+        self, discovery_info: Optional[Dict[str, Any]] = None
+    ) -> Dict[str, Any]:
+        """Handle a flow initialized by zeroconf discovery."""
+        name: str = discovery_info[CONF_NAME]
+        host: str = discovery_info[CONF_HOST]
+        bond_id = name.partition(".")[0]
+        await self.async_set_unique_id(bond_id)
+        self._abort_if_unique_id_configured({CONF_HOST: host})
+
+        self._discovered = {
+            CONF_HOST: discovery_info["host"],
+            "bond_id": bond_id,
```

Comment:
lets introduce a const for CONF_BOND_ID and use it in place of strings

Reply:
updated.